src/virt/emulator.rs

```diff
@@ -843,8 +868,7 @@ mod tests {
         assert_eq!(get_next_interrupt(0b000, 0b010, 0b000), None);
         assert_eq!(get_next_interrupt(0b010, 0b010, 0b010), None);
 
-        assert_eq!(get_next_interrupt(0b001, 0b001, 0b000), Some(0));
```

Comment:
This one we can keep, but yes the next one we have to update it with the new order.